Add area geometry support to runway preset

youssefelzedy opened the pull request:
## Add area geometry support to runway preset

### Summary
Adds support for area geometry to the runway preset to enable mapping of runway surface areas in addition to centerlines.

### Changes
- Added `"area"` to supported geometry types in runway preset
- Added `addTags` and `removeTags` configuration for proper tag management
- Maintains backward compatibility with existing line-based runway mapping
- Confirmed runway preset is already discoverable through existing area category (aeroway is included in area.json members)

### Motivation
Addresses issue #11169 which requested support for `area:aeroway=runway` tagging to distinguish between:
- **Line geometry**: Runway centerline (`aeroway=runway`)
- **Area geometry**: Runway surface area (`area:aeroway=runway`)

### Files Modified
- `data/presets/aeroway/runway.json` - Added area geometry support and tag management
- `data/preset_categories/area.json` - Verified aeroway category inclusion (no changes needed)

### Implementation Notes
This schema change provides the foundation for geometry-specific tagging. The actual logic to apply `area:aeroway=runway` for area geometries (vs `aeroway=runway` for lines) will need to be implemented in iD editor's JavaScript code, specifically in the `setTags` method of `modules/presets/preset.js`.

### Related
Fixes openstreetmap/iD#11169

youssefelzedy committed:
Add area geometry support to runway preset

- Add "area" to supported geometry types for runway surface mapping
- Include addTags and removeTags for proper tag cleanup
- Addresses openstreetmap/iD#11169 for area:aeroway=runway support

youssefelzedy committed:
Update preset defaults configuration

- Modify point preset defaults ordering
- Adjust vertex preset priority settings
- Update default preset categories for improved user experience

k-yle commented on data/preset_defaults.json:
```diff
@@ -9,6 +9,7 @@
         "amenity/place_of_worship",
         "amenity/cafe",
         "amenity/restaurant",
+        "aeroway/runway",
```
the purpose of this file is to define the default presets which are shown to new users who have no "recently selected" presets:


<img width="314" height="694" alt="image" src="https://github.com/user-attachments/assets/ff2a0b54-8913-474f-9ed6-3c0415b37054" />

So I think the changes to this file are unrelated to the goal of this PR?

k-yle commented on data/presets/aeroway/runway.json:
```diff
@@ -7,11 +7,19 @@
+    },
+    "removeTags": {
+        "aeroway": "*",
+        "area:aeroway": "*"
```
a simpler solution would be to leave the existing preset untouched, and create a single new file called `runway_area.json`.

[`pedestrian_line.json`](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/highway/pedestrian_line.json) and [`pedestrian_area.json`](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/highway/pedestrian_area.json) is a good example

youssefelzedy committed:
Split runway preset: add runway_area.json for area geometry

Separates runway preset into line (runway.json) and area (runway_area.json)
variants, following the pedestrian preset pattern.

youssefelzedy committed:
Remove line runway from preset_defaults.json

matkoniecz commented on data/preset_categories/area.json:
```diff
@@ -0,0 +1,10 @@
+{
```
@youssefelzedy ping?

youssefelzedy replied:
area.json is not adding a new tag or changing map rendering by itself. It defines a new preset category in the iD tagging schema so area-based presets can be grouped together in the editor UI.

In this PR, the category is needed so the new area-based aeroway preset, especially aeroway=runway with area:aeroway=*, has a proper place in the schema. Without this file, the preset can still exist, but it would not be organized under an Area category in iD, which makes it harder for users to discover and use.

- iD shows an Area category in the preset picker.
- Area-based aeroway presets, including runway areas, are grouped there.
- This helps users find and apply the correct area representation for aeroway=runway while keeping the existing aeroway=runway centerline behavior unchanged.

Sorry for the late response!

matkoniecz commented:
> @[youssefelzedy](https://github.com/youssefelzedy) youssefelzedy requested a review from [matkoniecz](https://github.com/matkoniecz) [6 hours ago](https://github.com/openstreetmap/id-tagging-schema/pull/1648#event-24732948621)

https://github.com/openstreetmap/id-tagging-schema/pull/1648#pullrequestreview-3682179543 remains unprocessed for last three months - why this file got added?

matkoniecz commented:
And https://github.com/openstreetmap/id-tagging-schema/pull/1648#issuecomment-3149884209 waits since Aug 4, 2025

youssefelzedy commented:
> [#1648 (review)](https://github.com/openstreetmap/id-tagging-schema/pull/1648#pullrequestreview-3682179543) remains unprocessed for the last three months - why was this file got added?

@matkoniecz The functional part is runway_area.json, which adds the area runway preset. area.json only affects preset grouping in the UI. 

If you prefer keeping this PR strictly scoped to runway-area support, I can drop the category file and keep only the runway preset change.

If you meant the Kyle review

I added aeroway/runway to preset_defaults.json because I thought runway should be easier to discover after adding area support, but I agree that this is a separate default-presets UX change and not part of the core area-runway support.
